### PR TITLE
fixes for "improve description of --initial-cluster-state flag" #15743 

### PR DESCRIPTION
**Pull request by sharathsivakumar**
Improves the description for --initial-clsuter-state provided in the help. Details of the issue can be found [here](https://github.com/etcd-io/etcd/issues/15743)

### CODE REVIEW

File: server/etcdmain/help.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ Clustering:
   --initial-cluster 'default=http://localhost:2380'
     Initial cluster configuration for bootstrapping.
   --initial-cluster-state 'new'
-    Initial cluster state ('new' or 'existing').
+    Initial cluster state ('new' or 'existing'). Initial cluster state is based on the availability of a populated data/WAL directory.
```
[the remark this answers is not in the document]
Comment (sharathsivakumar) [reply server/etcdmain/help.go]: @ahrtr I have built upon your suggestion. I have added the bootstrapping term and restoring a cluster as that explicitly creates the need for having WAL. And this is in line with the code and the way the fucntion works. Do let me know if this fine. I am bit concerned that this particular definition is bit long. What are your thoughts on this?

[the remark this answers is not in the document]
Comment (ahrtr) [reply server/etcdmain/help.go]: Suggest to remove `"restoring an existing cluster or`", when there is data (e.g. member/wal), then it doesn't matter what's the value for `--initial-cluster-state`

[the remark this answers is not in the document]
Comment (sharathsivakumar) [reply server/etcdmain/help.go]: @ahrtr Makes sense. I have made changes accordingly.

File: server/etcdmain/config.go [hunk 2]
```diff
@@ -189,8 +189,7 @@ func newConfig() *config {
 	fs.StringVar(&cfg.ec.DNSClusterServiceName, "discovery-srv-name", cfg.ec.DNSClusterServiceName, "Service name to query when using DNS discovery.")
 	fs.StringVar(&cfg.ec.InitialCluster, "initial-cluster", cfg.ec.InitialCluster, "Initial cluster configuration for bootstrapping.")
 	fs.StringVar(&cfg.ec.InitialClusterToken, "initial-cluster-token", cfg.ec.InitialClusterToken, "Initial cluster token for the etcd cluster during bootstrap.")
-	fs.Var(cfg.cf.clusterState, "initial-cluster-state", "Initial cluster state ('new' or 'existing').")
-
+	fs.Var(cfg.cf.clusterState, "initial-cluster-state", "Initial cluster state ('new' when bootstrapping a new cluster or 'existing' when adding new members to an existing cluster).Initial cluster state is based on the availability of a populated data/WAL directory.")
```
Comment (ahrtr) [inline server/etcdmain/config.go:192]: Suggest to remove "`Initial cluster state is based on the availability of a populated data/WAL directory.`". Note that there shouldn't have local data no matter bootstrapping a new cluster or adding new members into an existing cluster. 

After successfully bootstrapping the new cluster or adding any new members, then it doesn't matter what's the value for the flag.

**Comment by ahrtr** [comment]
Please also squash the commits.

File: server/etcdmain/config.go (hunk 2 above)
Comment (lavacat) [reply server/etcdmain/config.go:192]: How about we also add `After successful initialization (bootstrapping or adding), flag is ignored on restarts`

Comment (ahrtr) [reply server/etcdmain/config.go:192]: Agreed.

Comment (sharathsivakumar) [reply server/etcdmain/config.go]: done!

Comment (sharathsivakumar) [reply server/etcdmain/config.go]: @ahrtr and @lavacat, squashed the commits and updated the desc based on the review inputs

File: server/etcdmain/config.go [hunk 3]
```diff
@@ -189,8 +189,7 @@ func newConfig() *config {
 	fs.StringVar(&cfg.ec.DNSClusterServiceName, "discovery-srv-name", cfg.ec.DNSClusterServiceName, "Service name to query when using DNS discovery.")
 	fs.StringVar(&cfg.ec.InitialCluster, "initial-cluster", cfg.ec.InitialCluster, "Initial cluster configuration for bootstrapping.")
 	fs.StringVar(&cfg.ec.InitialClusterToken, "initial-cluster-token", cfg.ec.InitialClusterToken, "Initial cluster token for the etcd cluster during bootstrap.")
-	fs.Var(cfg.cf.clusterState, "initial-cluster-state", "Initial cluster state ('new' or 'existing').")
-
+	fs.Var(cfg.cf.clusterState, "initial-cluster-state", "Initial cluster state ('new' when bootstrapping a new cluster or 'existing' when adding new members to an existing cluster).After successful initialization (bootstrapping or adding), flag is ignored on restarts.")
```
Comment (jmhbnz) [inline server/etcdmain/config.go:192]: Nitpick, I think there is a space missing here.

```suggestion
	fs.Var(cfg.cf.clusterState, "initial-cluster-state", "Initial cluster state ('new' when bootstrapping a new cluster or 'existing' when adding new members to an existing cluster). After successful initialization (bootstrapping or adding), flag is ignored on restarts.")
```

**Review by jmhbnz — APPROVED** [review]
Overall this looks good to me, thanks @sharathsivakumar.

Just one minor nitpick about a potentially missing space after a sentence finishes.

File: server/etcdmain/config.go (hunk 3 above)
Comment (sharathsivakumar) [reply server/etcdmain/config.go]: Thanks for that. Fixed it.

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @sharathsivakumar

**Comment by ahrtr** [comment]
@sharathsivakumar could you please also backport this PR to 3.5 and 3.4? thx

**Pull request merged**